client/register.go

```diff
@@ -43,11 +43,20 @@ var registerTimeout = cmdRegister.Flag.Int("t", 5, "")
 const registerRecheckTime = 10 * time.Millisecond
 
 func runRegister(cmd *Command, args []string) {
-	if *registerKey == "" && *registerKeyFile == "" {
+	k := NewKeysFile(daemonFolder + daemonToRegister)
+	if *registerRemove && *registerKey == "" && *registerKeyFile == "" {
+		// Short circuit & handle `knox register -r`, which is expected to remove all keys
+		err := k.Overwrite([]string{})
```

Comment:
I think we need a call to Lock before this

Reply:
oops, my bad